src/btree/bt_curnext.c

```diff
@@ -955,7 +956,7 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
     case 0:
         F_SET(cursor, WT_CURSTD_KEY_INT | WT_CURSTD_VALUE_INT);
 
-        if (DIAGNOSTIC_ASSERTS_ENABLED(session)) {
+        if (DIAGNOSTIC_ASSERTS_ENABLED(session, WT_DIAG_OUT_OF_ORDER)) {
```

Comment:
Are we planning to have two separate categories for keys out of orders vs timestamps out of order. If they are rolled into one, would a category implying "check corruption" feel more intuitive.